src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs

```diff
@@ -1962,43 +1955,22 @@ class NotDisposable
                         IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'b')
                       Right: 
                         ILiteralOperation (OperationKind.Literal, Type: System.Boolean, Constant: True) (Syntax: 'true')
-
-            Next (Regular) Block[B8]
+           Next (Regular) Block[B6]
                 Finalizing: {R5}
                 Leaving: {R4} {R3} {R1}
     }
     .finally {R5}
     {
         Block[B5] - Block
             Predecessors (0)
-            Statements (0)
-            Jump if True (Regular) to Block[B7]
```

Comment:
>Jump if True (Regular) to Block[B7] [](start = 12, length = 35)

I would be more comfortable with the change if it didn't have any observable effect on scenarios that weren't actually broken before. Please consider restoring original error recovery logic.

Reply:
I think I would argue that this *was* broken though. We were returning a CFG that said we would call `IDisposable.Dispose()` on something that we know isn't disposable.

Reply:
> I think I would argue that this was broken though. We were returning a CFG that said we would call IDisposable.Dispose() on something that we know isn't disposable.

This is an error recovery and the IInvocationOperation actually has all the right types in place, receiver is System.IDisposable. There is a conversion that would fail at runtime, but I don't see any harm in its presence because IConversionOperation doesn't imply that conversion succeeds. In fact, we even say that it is IsInvalid. I would argue that  the graph actually reflect user's intent as compiler "understood" it.

---
In reply to: [528057465](https://github.com/dotnet/roslyn/pull/49481#discussion_r528057465) [](ancestors = 528057465)

Reply:
Makes sense. I've restored it back to how it was. The only changes now are for IAsync disposal which was clearly incorrect.